src/test/java/com/google/api/generator/gapic/utils/SampleCodeJavaFormatterTest.java

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.utils;
+
+import static junit.framework.TestCase.assertEquals;
+
+import com.google.api.generator.engine.ast.AssignmentExpr;
+import com.google.api.generator.engine.ast.Expr;
+import com.google.api.generator.engine.ast.ExprStatement;
+import com.google.api.generator.engine.ast.MethodInvocationExpr;
+import com.google.api.generator.engine.ast.PrimitiveValue;
+import com.google.api.generator.engine.ast.Statement;
+import com.google.api.generator.engine.ast.TryCatchStatement;
+import com.google.api.generator.engine.ast.TypeNode;
+import com.google.api.generator.engine.ast.Value;
+import com.google.api.generator.engine.ast.ValueExpr;
+import com.google.api.generator.engine.ast.VaporReference;
+import com.google.api.generator.engine.ast.Variable;
+import com.google.api.generator.engine.ast.VariableExpr;
+import com.google.api.generator.engine.writer.JavaWriterVisitor;
+import com.google.api.generator.gapic.utils.SampleCodeJavaFormatter.FormatException;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+
+public class SampleCodeJavaFormatterTest {
+
+  @Test
+  public void validFormatSampleCode_tryCatchStatement() {
+    List<Statement> statements = Arrays.asList(createTryCatchSampleCode());
+    String result = SampleCodeJavaFormatter.format(writeStatements(statements));
+    String expected =
+        String.format(createLines(3), "try (boolean condition = false) {\n", "  int x = 3;\n", "}");
+    assertEquals(expected, result);
+  }
+
+  @Test
+  public void validFormatSampleCode_longLineStatement() {
```

Comment:
Since we're just testing the formatter, let's simplify this test by directly writing strings instead of constructing code via the AST.